src/response.rs

```diff
@@ -491,32 +491,26 @@ impl AsMut<Headers> for Response {
     }
 }
 
-impl From<Response> for Body {
-    fn from(res: Response) -> Body {
-        res.body
+impl From<()> for Response {
+    fn from(_: ()) -> Self {
+        Response::new(StatusCode::NoContent)
     }
 }
 
-impl From<String> for Response {
-    fn from(s: String) -> Self {
-        let mut res = Response::new(StatusCode::Ok);
-        res.set_body(s);
-        res
-    }
-}
-
-impl<'a> From<&'a str> for Response {
-    fn from(s: &'a str) -> Self {
-        let mut res = Response::new(StatusCode::Ok);
-        res.set_body(s);
-        res
+impl From<StatusCode> for Response {
+    fn from(s: StatusCode) -> Self {
+        Response::new(s)
     }
 }
 
-impl From<Vec<u8>> for Response {
-    fn from(b: Vec<u8>) -> Self {
+impl<T> From<T> for Response
+where
+    T: Into<Body>,
+{
+    fn from(value: T) -> Self {
+        let body: Body = value.into();
         let mut res = Response::new(StatusCode::Ok);
-        res.set_body(b);
+        res.set_body(body);
```

Comment:
This is a minor tweak, but this could be 

```rust
impl<T> From<T> for Response
where
    T: Into<Body>,
{
    fn from(value: T) -> Self {
        let mut res = Response::new(StatusCode::Ok);
        res.set_body(value);
        res
    }
}
```

since replace_body does the `.into::<Body>()`

meta: is this sort of review useful?

Reply:
@jbr thanks heaps for suggesting this; yeah this is very useful!